packages/boot/test/bootstrapTests/test-net-ibc-upgrade.ts

```diff
@@ -111,66 +111,68 @@ test.serial('upgrade at many points in network API flow', async t => {
+      const serverPort = await EV(networkVat).bind('/ibc-port/');
+      const clientPort = await EV(networkVat).bind('/ibc-port/');
+      return [label, { serverPort, clientPort }];
+    },
+    startServer: async ([label, opts]) => {
       const started = await EV(zoe).startInstance(
         installation.ibcServerMock,
         {},
         {},
-        { boundPort },
+        { boundPort: opts.serverPort },
       );
-      t.truthy(started.creatorFacet);
-      return { serverPort: boundPort, server: started.creatorFacet };
+      t.truthy(started.creatorFacet, `${label} ibcServerMock`);
+      return [label, { ...opts, server: started.creatorFacet }];
     },
-    startListening: async opts => {
+    startListening: async ([label, opts]) => {
       await EV.sendOnly(opts.server).listen();
-      return opts;
-    },
-    getAddress: async opts => {
-      const address = await EV(opts.serverPort).getLocalAddress();
-      t.log('server address', address);
-      return { ...opts, address };
+      await EV.sendOnly(opts.server).dequeue('onListen');
+      return [label, opts];
     },
-    allocatePort: async opts => {
-      const clientPort = await EV(networkVat).bind('/ibc-port/');
-      return { ...opts, clientPort };
-    },
-    getClientAddress: async opts => {
-      const address = await EV(opts.clientPort).getLocalAddress();
-      t.log('client address', address);
-      return opts;
+    getAddresses: async ([label, opts]) => {
+      const serverAddress = await EV(opts.serverPort).getLocalAddress();
+      const clientAddress = await EV(opts.clientPort).getLocalAddress();
+      t.log(`${label} server ${serverAddress} client ${clientAddress}`);
+      return [label, { ...opts, serverAddress }];
     },
-    startClient: async opts => {
+    startClient: async ([label, opts]) => {
       const started = await EV(zoe).startInstance(
         installation.ibcClientMock,
         {},
         {},
         { myPort: opts.clientPort },
       );
-      t.truthy(started.creatorFacet);
-      return { ...opts, client: started.creatorFacet };
+      t.truthy(started.creatorFacet, `${label} ibcClientMock`);
+      return [label, { ...opts, client: started.creatorFacet }];
+    },
+    startConnecting: async ([label, opts]) => {
+      await EV.sendOnly(opts.client).connect(opts.serverAddress);
+      await EV.sendOnly(opts.server).dequeue('onAccept');
+      await EV.sendOnly(opts.server).dequeue('onOpen');
+      return [label, opts];
     },
-    startConnecting: async opts => {
-      await EV.sendOnly(opts.client).connect(opts.address);
-      return opts;
+    sendPacket: async ([label, opts]) => {
+      await EV.sendOnly(opts.client).send(label);
+      return [label, opts];
     },
-    sendPacket: async opts => {
-      await EV.sendOnly(opts.client).send('hello198');
-      return opts;
+    respond: async ([label, opts]) => {
+      await EV.sendOnly(opts.server).dequeue('onReceive');
+      return [label, opts];
     },
```

Comment:
@dckc Having this `dequeue('onReceive')` in an upgrade-interruptible step after `.send(…)` introduces the kind of test failure you were expecting. I think the same kind of separation should be used for all `dequeue`s, and possibly something similar will be needed in the other mock implementations.